nixos/modules/services/hardware/monado.nix

```
@@ -0,0 +1,93 @@
+{ config
+, lib
+, pkgs
+, ...
+}:
+let
+  inherit (lib) mkDefault mkEnableOption mkIf mkOption mkPackageOption types;
+
+  cfg = config.services.monado;
+
+in
+{
+  options.services.monado = {
+    enable = mkEnableOption "Monado wrapper and user service";
+
+    package = mkPackageOption pkgs "monado" { };
+
+    defaultRuntime = mkOption {
+      type = types.bool;
+      description = ''
+        Whether to enable Monado as the default OpenXR runtime on the system.
+
+        Note that applications can bypass this option by setting an active
+        runtime in a writable XDG_CONFIG_DIRS location like `~/.config`.
+      '';
+      default = false;
+      example = true;
+    };
+  };
+
+  config = mkIf cfg.enable {
+    security.wrappers."monado-service" = {
+      setuid = false;
+      owner = "root";
+      group = "root";
+      # cap_sys_nice needed for asynchronous reprojection
+      capabilities = "cap_sys_nice+eip";
+      source = "${cfg.package}/bin/monado-service";
+    };
+
+    services.udev.packages = with pkgs; [ xr-hardware ];
+
+    systemd.user = {
+      services.monado = {
+        description = "Monado XR runtime service module";
+        requires = [ "monado.socket" ];
+        conflicts = [ "monado-dev.service" ];
+
+        unitConfig.ConditionUser = "!root";
+
+        environment = {
+          # Default options
+          # https://gitlab.freedesktop.org/monado/monado/-/blob/4548e1738591d0904f8db4df8ede652ece889a76/src/xrt/targets/service/monado.in.service#L12
+          XRT_COMPOSITOR_LOG = mkDefault "debug";
+          XRT_PRINT_OPTIONS = mkDefault "on";
+          IPC_EXIT_ON_DISCONNECT = mkDefault "off";
+        };
+
+        serviceConfig = {
+          ExecStart = "${config.security.wrapperDir}/monado-service";
+          Restart = "no";
```

> I am not sure how to reproduce this

I'm not sure how I even produced this anymore but this should fix the issue:

```suggestion
          Restart = "on-failure";
```

---

This would deviate from the upstream systemd user service unit.

I am not quite sure why your service was restarted so quickly, when `Restart` is `no`.

---

Okay I could reproduce this issue!

It happens when Monado fails to start, while an OpenXR app tries to connect to the socket. I remember that there was some option for this in `systemd.socket` to prevent socket messages from rapidly issuing start requests